src/apify_client/clients/resource_clients/log.py

```diff
@@ -378,3 +382,160 @@ async def _stream_log(self) -> None:
+    # final message will be captured, but increases the chances of that.
+    _final_sleep_time_s = 6
+
+    def __init__(self, *, to_logger: logging.Logger, check_period: timedelta = timedelta(seconds=5)) -> None:
+        """Initialize `StatusMessageWatcher`.
+
+        Args:
+            to_logger: The logger to which the status message will be redirected.
+            check_period: The period with which the status message will be polled.
+        """
+        if self._force_propagate:
+            to_logger.propagate = True
+        self._to_logger = to_logger
+        self._check_period = check_period.total_seconds()
+        self._last_status_message = ''
+
+    def _log_run_data(self, run_data: dict[str, Any] | None) -> bool:
+        """Get relevant run data, log them if changed and return `True` if more data is expected.
+
+        Args:
+            run_data: The dictionary that contains the run data.
+
+        Returns:
+              `True` if more data is expected, `False` otherwise.
+        """
+        if run_data is not None:
+            status = run_data.get('status', 'Unknown status')
+            status_message = run_data.get('statusMessage', '')
+            new_status_message = f'Status: {status}, Message: {status_message}'
+
+            if new_status_message != self._last_status_message:
+                self._last_status_message = new_status_message
+                self._to_logger.info(new_status_message)
+
+            return not (run_data.get('isStatusMessageTerminal', False))
+        return True
+
+
+class StatusMessageWatcherAsync(StatusMessageWatcher):
+    """Async variant of `StatusMessageWatcher` that is logging in task."""
+
+    def __init__(
+        self, *, run_client: RunClientAsync, to_logger: logging.Logger, check_period: timedelta = timedelta(seconds=1)
+    ) -> None:
+        """Initialize `StatusMessageWatcherAsync`.
+
+        Args:
+            run_client: The client for run that will be used to get a status and message.
+            to_logger: The logger to which the status message will be redirected.
+            check_period: The period with which the status message will be polled.
+        """
+        super().__init__(to_logger=to_logger, check_period=check_period)
+        self._run_client = run_client
+        self._logging_task: Task | None = None
+
+    def start(self) -> Task:
+        """Start the logging task. The caller has to handle any cleanup by manually calling the `stop` method."""
+        if self._logging_task:
+            raise RuntimeError('Logging task already active')
+        self._logging_task = asyncio.create_task(self._log_changed_status_message())
+        return self._logging_task
+
+    def stop(self) -> None:
+        """Stop the logging task."""
+        if not self._logging_task:
+            raise RuntimeError('Logging task is not active')
+
+        self._logging_task.cancel()
```

Comment:
I'm afraid there might be GC-related warnings if you don't await the task ([docs](https://docs.python.org/3/library/asyncio-task.html#asyncio.Task.cancel))